File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/DomainRecordDao.java
package com.platdmit.simplecloudmanager.data.database.dao;

import com.platdmit.simplecloudmanager.data.database.entity.DbDomainRecord;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface DomainRecordDao extends BaseDao<DbDomainRecord>{
    @Query("SELECT * FROM dbdomainrecord WHERE mParentDomain = :id")
    List<DbDomainRecord> getRecordsForDomain(long id);

    @Query("SELECT * FROM dbdomainrecord WHERE mId = :id")
    DbDomainRecord getRecord(long id);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/DomainRecordDao.kt
package com.platdmit.simplecloudmanager.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.platdmit.simplecloudmanager.data.database.entity.DbDomainRecord

@Dao
interface DomainRecordDao : BaseDao<DbDomainRecord> {
    @Query("SELECT * FROM dbdomainrecord WHERE parentDomain = :id")
    fun getRecordsForDomain(id: Long): List<DbDomainRecord>?

    @Query("SELECT * FROM dbdomainrecord WHERE id = :id")
    fun getRecord(id: Long): DbDomainRecord?
}